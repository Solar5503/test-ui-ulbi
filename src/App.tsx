// import { useEffect, useState } from 'react'
import './App.css'
import Navbar from './components/Navbar/Navbar'
import AppRouter from './router/AppRouter'

function App() {
  // const [data, setData] = useState<null | object>(null)
  // const [toggle, setToggle] = useState<boolean>(false)
  // const [value, setValue] = useState<string>('')
  // const clickHandler = () => setToggle((prev) => !prev)
  // useEffect(() => {
  //   setTimeout(() => {
  //     setData({})
  //   }, 100)
  // }, [])
  // return (
  //   <div>
  //     <h1 data-testid="value-elem">{value}</h1>
  //     {toggle && <div data-testid="toggle-elem">toggle</div>}
  //     {data && <div style={{ color: 'red' }}>data</div>}
  //     <h1>Hello world</h1>
  //     <button data-testid="toggle-btn" onClick={clickHandler}>
  //       Click me
  //     </button>
  //     <input
  //       type="text"
  //       placeholder="input value...."
  //       onChange={(e) => setValue(e.target.value)}
  //       value={value}
  //     />
  //   </div>
  // )

  return (
    <>
      <Navbar />
      <AppRouter />
    </>
  )
}

export default App
