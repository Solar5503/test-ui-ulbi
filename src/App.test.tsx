import '@testing-library/jest-dom'
import { fireEvent, render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import App from './App'

describe('Test App', () => {
  test('render header', () => {
    render(<App />)
    const header = screen.getByText(/hello world/i)
    const btn = screen.getByRole('button')
    const input = screen.getByPlaceholderText(/input value/i)
    expect(header).toBeInTheDocument()
    expect(btn).toBeInTheDocument()
    expect(input).toBeInTheDocument()
    // screen.debug()
    expect(input).toMatchSnapshot()
  })
  test('not element', () => {
    render(<App />)
    const header = screen.queryByText(/hello2/i)
    expect(header).toBeNull()
  })
  test('async element', async () => {
    render(<App />)
    screen.debug()
    const header = await screen.findByText(/data/i)
    expect(header).toBeInTheDocument()
    expect(header).toHaveStyle({ color: 'red' })
    screen.debug()
  })
  test('click event', () => {
    render(<App />)
    const btn = screen.getByTestId('toggle-btn')
    expect(screen.queryByTestId('toggle-elem')).toBeNull()
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeInTheDocument()
    fireEvent.click(btn)
    expect(screen.queryByTestId('toggle-elem')).toBeNull()
  })
  test('input event', async () => {
    const user = userEvent.setup()
    render(<App />)
    const input = screen.getByPlaceholderText(/input value/i)
    expect(screen.queryByTestId('value-elem')).toContainHTML('')
    // fireEvent.input(input, { target: { value: '123123' } })
    await user.type(input, '123123')
    expect(screen.queryByTestId('value-elem')).toContainHTML('123123')
  })
})
