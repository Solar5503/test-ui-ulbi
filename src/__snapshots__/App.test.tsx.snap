// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test App render header 1`] = `
<input
  placeholder="input value...."
  type="text"
/>
`;
